website/docs/docs/building-a-dbt-project/hooks-operations.md

```diff
@@ -27,14 +27,117 @@ Effective database administration sometimes requires additional SQL statements t
 
 dbt provides two different interfaces for you to version control and execute these statements as part of your dbt project — hooks and operations.
 
-### Hooks
+## About hooks
+
 Hooks are snippets of SQL that are executed at different times:
   * `pre-hook`: executed _before_ a model, seed or snapshot is built.
   * `post-hook`: executed _after_ a model, seed or snapshot is built.
   * `on-run-start`: executed at the _start_ of `dbt run`, `dbt seed` or `dbt snapshot`
   * `on-run-end`: executed at the _end_ of `dbt run`, `dbt seed` or `dbt snapshot`
 
-Hooks are defined in your `dbt_project.yml` file. Pre- and post-hooks can also be defined in a `config` block.
+Hooks can help you leverage database-specific actions to expand dbt.
+
+You can define hooks in your `dbt_project.yml` file. Pre- and post-hooks can also be defined in a `config` block.
+
+<VersionBlock firstVersion="1.2">
+
+In order to streamline hooks and automatically apply grants when your dbt model runs, we recommend using [`grants` resource-config](/reference/resource-configs/grants).  
+
+### Examples using hooks
+
+You can use hooks to trigger actions at certain times when running an operation or building a model, seed, or snapshot.
+
+For more information about when hooks can be triggered, see reference sections for [`on-run-start` and `on-run-end` hooks](on-run-start-on-run-end) and [`pre-hook`s and `post-hook`s](pre-hook-post-hook).
+
+You can use hooks to provide database-specific functionality not available out-of-the-box with dbt. 
+
+For example, you can use a `config` block to run an `ALTER TABLE` statement right after building an individual model using a `post-hook`:
+
+<WHCode>
+
+<div warehouse="BigQuery">
+
+<File name='models/<model_name>.sql'>
+
+```sql
+{{ config(
+    post_hook=[
+      "grant select on {{ this }} to role reporter"
```

Comment:
BigQuery has different names for its privileges + grantees. So if you're writing the hook SQL yourself, this actually needs to look like:
```sql
'grant `roles/bigquery.dataViewer` on {{ this.type }} {{ this }} to "user:someone@yourcompany.com"'
```
If you're using the new-in-1.2 grants config, it will look like:
```yml
{{ config(grants = {'roles/bigquery.dataViewer': ['user:someone@yourcompany.com']}) }}
```

**This is relevant to our `grants` docs, too.** We need to call out that, while our examples use words like `select` and `another_user` because they're common on many databases, users will need to be aware of their own database's support syntax for:
- **privileges** (= a thing you can do in the database)
- **grantees** (= the recipients of granted privileges, also called "principals") — which could be a user, a group of users, a role held by users (Snowflake), a service account (GCP), ... 

(Very confusing: Google uses the word "role" to mean "privilege," whereas Snowflake uses the word "role" to mean "grantee" = a way of receiving privileges. Let's just try avoiding the word "role" if possible!)

Reply:
BigQuery distinction addressed in https://github.com/dbt-labs/docs.getdbt.com/pull/1642/commits/f4c9578e38f22addf04835bcc9e0de536aef8528

Overall examples note addressed here: https://github.com/dbt-labs/docs.getdbt.com/pull/1642/commits/8af57ee062cc36b91d3fad081d2969e723dc50eb